### DISCUSSION TITLE
Why there is iced and egui ?

### DISCUSSION THREAD

**Discussion by ghost** [hecrj/iced#1478] [General]
Hi, i was wondering, why there are 2 GUI libraries iced and egui, why don't the developers of those 2 libraries join forces and develop only one library ?

**Comment by Malfeasant** [comment]
Seeing as egui is [immediate mode](https://en.wikipedia.org/wiki/Immediate_mode_GUI), and iced is [retained mode](https://en.wikipedia.org/wiki/Retained_mode), the answer is that they use completely different approaches.  Sometimes you want one, sometimes you want the other.
